contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+from opencensus.ext.azure.common import Options
+from opencensus.metrics import transport
+from opencensus.stats import stats
+
+__all__ = ['MetricsExporter', 'new_metrics_exporter']
+
+
+class MetricsExporter(object):
+    """Metrics exporter for Microsoft Azure Monitor."""
+
+    def __init__(self, options):
+        self._options = options
+        self._md_cache = {}
+        self._md_lock = threading.Lock()
+
+    def export_metrics(self, metrics):
+        metrics = list(metrics)
+        for metric in metrics:
+            # TODO: implement export to Azure Monitor functionality
+            print(repr(metric))
+
+
+def new_metrics_exporter(**options):
+    options = Options(**options)
```

Comment:
I'm curious why would we want to validate options here versus in the `MetricsExporter.__init__`.

Reply:
I feel like validation logic should be handled in the service layer, outside of the constructor. IMO init should only serve to instantiate variables.

Reply:
Would you explain a bit more?
In general people do validate inside constructors, which prevent them from creating a wrong object.

https://github.com/googleapis/google-cloud-python/blob/master/monitoring/google/cloud/monitoring_v3/gapic/metric_service_client.py#L115

Reply:
Hmm, it does seem like we do validation everywhere inside the constructor for a lot of classes. I can make the change to be consistent.